AnkiDroid/src/main/java/com/ichi2/anki/SharedDecksDownloadFragment.kt

```
@@ -0,0 +1,447 @@
+                        Timber.i("Download could not be successful, update UI and unregister receiver")
+                        Timber.d("Status code -> ${it.getInt(columnIndex)}")
+                        checkDownloadStatusAndUnregisterReceiver(isSuccessful = false)
+                        return
+                    }
+                }
+            }
+
+            try {
+                verifyDeckIsImportable()
+            } catch (exception: Exception) {
+                Timber.w(exception)
+                checkDownloadStatusAndUnregisterReceiver(isSuccessful = false)
+                return
+            }
+
+            if (isVisible) {
+                // Setting these since progress checker can stop before progress is updated to represent 100%
+                mDownloadPercentageText.text = getString(R.string.percentage, DOWNLOAD_COMPLETED_PROGRESS_PERCENTAGE)
+                mDownloadProgressBar.progress = DOWNLOAD_COMPLETED_PROGRESS_PERCENTAGE.toInt()
+
+                // Remove cancel button and show import deck button
+                mCancelButton.visibility = View.GONE
+                mImportDeckButton.visibility = View.VISIBLE
+            }
+
+            Timber.i("Opening downloaded deck for import")
+            openDownloadedDeck(context)
+
+            Timber.d("Checking download status and unregistering receiver")
+            checkDownloadStatusAndUnregisterReceiver(isSuccessful = true)
+        }
+    }
+
+    /**
+     * Unregister the mOnComplete broadcast receiver.
+     */
+    private fun unregisterReceiver() {
+        Timber.d("Unregistering receiver")
+        // This might throw an exception in cases where the receiver is already in unregistered state.
+        // Log the exception in such cases, there is nothing else to do.
+        try {
+            activity?.unregisterReceiver(mOnComplete)
+        } catch (e: Exception) {
+            Timber.w(e)
```

move the comment here, and be a bit more explicit that the exception is ignored. Maybe with a `return`